### ISSUE TITLE
Specific file type upload fails on Gradio 5

### ISSUE THREAD

**Issue by FrostMegaByte** [gradio-app/gradio#9752]
### Describe the bug

I added code that allows uploading an Excel file. This works on Gradio version 4.44.1, but not on version 5.

On version 5 an error pop-up appears that mentions that the file upload for the specific file type failed. I uploaded an Excel file and then the error mentions something like only [".xlsx"] files are allowed, which I did upload.

When setting the file types to ["files"] the upload succeeds, but that defeats the purpose of being able to specify specific types.

### Have you searched existing issues?  🔎

- [X] I have searched and found no existing issues

### Reproduction

```python
import gradio as gr

file_output = gr.File()
file = gr.UploadButton(
	"Upload Excel file",
	file_count="single",
	file_types=[".xlsx"],
)
file.upload(lambda file: file.name, file, file_output)
```

### System Info

```shell
Windows 10
gradio==5.1.0
python==3.11
```

### Severity

I can work around it

**Comment by abidlabs** [comment]
Hi @FrostMegaByte I believe this is fixed on `main`, you can install Gradio from `main` like this:

```
pip install https://gradio-pypi-previews.s3.amazonaws.com/279fcf35468985e712835344e5a96ba3cd815609/gradio-5.1.0-py3-none-any.whl
```

would you be able to confirm?

**Comment by dogus-dursun** [comment]
> Hi @FrostMegaByte I believe this is fixed on `main`, you can install Gradio from `main` like this:
> 
> ```
> pip install https://gradio-pypi-previews.s3.amazonaws.com/279fcf35468985e712835344e5a96ba3cd815609/gradio-5.1.0-py3-none-any.whl
> ```
> 
> would you be able to confirm?

I have the exact same issue working with ".docx" files. I try to pip install Gradio from main with the given link, but it raises an error telling that it requires gradio-client-1.4.1, which is not available

**Comment by freddyaboulton** [comment]
Just verified the xlsx issue is fixed on 5.4.0!

**Comment by abidlabs** [comment]
Hi @Philrobots feel free to open a new issue with a repro
